From: Jeff Mahoney <jeffm@suse.com>
Subject: ocfs2: fix -Wformat warnings when building UML on x86-64

 The check to determine which format string is appopriate for u64 and
 friends works in most cases, but UML on x86_64 doesn't define CONFIG_X86_64,
 so it results in screen fulls of compile-time warnings.

 This patch fixes it to handle that case.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
--- linux-2.6.15/fs/ocfs2/cluster/masklog.h	2006-02-06 20:29:22.000000000 -0500
+++ linux-2.6.15-ocfs2/fs/ocfs2/cluster/masklog.h	2006-02-06 20:29:29.000000000 -0500
@@ -255,7 +255,7 @@
 	}								\
 } while (0)
 
-#if (BITS_PER_LONG == 32) || defined(CONFIG_X86_64)
+#if (BITS_PER_LONG == 32) || defined(CONFIG_X86_64) || (defined(CONFIG_UML_X86) && defined(CONFIG_64BIT))
 #define MLFi64 "lld"
 #define MLFu64 "llu"
 #define MLFx64 "llx"
